/*
 * Copyright contributors to Besu.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
package org.hyperledger.besu.sila.trie.pathbased.bonsai.code;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Module for providing the BonsaiCodeCache instance. This module is used to inject a singleton
 * instance of BonsaiCodeCache into the application.
 */
@Module
public class BonsaiCodeCacheModule {

  /** Creates a new instance of BonsaiCodeCacheModule. */
  public BonsaiCodeCacheModule() {
    // Default constructor
  }

  /**
   * Provides a singleton instance of BonsaiCodeCache.
   *
   * @return a new instance of BonsaiCodeCache
   */
  @Provides
  @Singleton
  public BonsaiCodeCache provideCodeCache() {
    return new BonsaiCodeCache();
  }
}
